stdlib/arg.ml

```diff
@@ -162,17 +167,17 @@ let parse_and_expand_argv_dynamic_aux allow_expand current argv speclist anonfun
           try
             let keyword, arg = split s in
             assoc3 keyword !speclist, Some arg
-          with Not_found -> stop (Unknown s)
+          with Not_found -> raise (convert_error (Unknown s))
       in
       let no_arg () =
         match follow with
         | None -> ()
-        | Some arg -> stop (Wrong (s, arg, "no argument")) in
+        | Some arg -> raise (Stop (Wrong (s, arg, "no argument"))) in
```

Comment:
I have simplified the code by moving it around and fusionning the different try clause in the main while loop.
Conversion now happens only at one point.